pype/plugins/global/publish/collect_rendered_files.py

```diff
@@ -66,14 +73,41 @@ def _process_path(self, data):
             os.environ["FTRACK_SERVER"] = ftrack["FTRACK_SERVER"]
 
         # now we can just add instances from json file and we are done
-        for instance in data.get("instances"):
+        for instance_data in data.get("instances"):
             self.log.info("  - processing instance for {}".format(
-                instance.get("subset")))
-            i = self._context.create_instance(instance.get("subset"))
-            self.log.info("remapping paths ...")
-            i.data["representations"] = [PypeLauncher().path_remapper(
-                data=r) for r in instance.get("representations")]
-            i.data.update(instance)
+                instance_data.get("subset")))
+            instance = self._context.create_instance(
+                instance_data.get("subset")
+            )
+            self.log.info("Filling stagignDir...")
+            instance.data.update(instance_data)
+
+            representations = []
+            for repre_data in instance_data.get("representations") or []:
+                staging_dir = repre_data.get("stagingDir")
+                if not staging_dir:
+                    pass
+
+                elif "{root" in staging_dir:
```

Comment:
this looks a bit iffy, can't we figure it out in some more elegant way than hardcoding `{root`?

Reply:
I can move this logic to `path_remapper` in Roots where hardcoding is acceptable because there is expected `"root"` key.